src/language/records.md

```diff
@@ -0,0 +1,142 @@
+---
+title: Records
+description: Summary of the record data structure in Dart.
+toc: false
+---
+
+Records are an anonymous, immutable, aggregate type. Like other [collection types][], 
+they let you bundle multiple objects into a single object. Unlike other collection 
+types, records are fixed-sized, heterogeneous, and typed.
+
+Records are real values; you can store them in variables, pass them to and from
+functions, and store them in lists.
+
+## Record syntax
+
+_Records expressions_ are comma-delimited lists of named or positional fields,
+enclosed in parenthesis:
+
+```dart
+var record = ('first', a: 2, b: true, 'last');    // Record expression
+```
+
+Named fields inside _record type annotations_ go inside a brace-delimited section
+of type and name pairs, after all positional fields:
+
+```dart
+(String, String?, {int a, bool b}) record;    // Record type annotation
+```
+Naming fields in record type annotations is optional for documentation purposes,
+but the names of named fields are not optional. For more information,
```

Comment:
I think it means, once given a name in annotation, the name can be omitted but it can't be replaced. Changed this to

```
Naming fields in record type annotations is optional for documentation purposes.
The names of those named fields are not optional, though,  
once you create a record with named fields. The field names become part of the
[record's type definition](#record-types). They can be omitted, but not replaced.

For more information, see the [Record equality](#record-equality) section.
```